src/util/request-manager.js

```diff
@@ -146,7 +146,13 @@ export default class RequestManager {
     }
 
     if (opts.httpsProxy != null) {
-      this.httpsProxy = opts.httpsProxy;
+      // fallback to httpProxy if httpsProxy is undefined (not specified).
+      // setting httpsProxy to `false` will not use a proxy for HTTPS. Requested by #4546
+      if (this.httpsProxy === undefined) {
```

Comment:
`this.httpsProxy` can never be `undefined` as far as I can trace this code. The constructor sets it to `null` and then `setOption` gets called either with `false` an empty string or a string containing the value. I am very confused at this point.